blockchain.go

```diff
@@ -1219,3 +1251,209 @@ func (b *Blockchain) testAlternativeHashAlgo(sig flowgo.TransactionSignature, ms
 
 	return nil
 }
+
+var _ runtime.Interface = &emulatorEnv{}
+
+// emulatorEnv implements runtime.Interface solely for the purpose of extracting state from a delta
+type emulatorEnv struct {
```

Comment:
I was a bit confused at first about all the missing implementations – maybe rename to something more specific, e.g. ``storageInspectionInterface`, `storageInspectionRuntimeInterface`, etc. that indicates this is only used for storage inspection, not for e.g. transaction or script execution

Reply:
good point

Reply:
why do we need this btw? I think here this doesn't make any sense

Reply:
The runtime interface implementation? You need it since the runtime context requires it to pass it along and I can't access the one from the FVM.

Reply:
Something like this? 

```go
	env, _ := fvm.NewTransactionEnvironment(
		b.vmCtx,
		b.vm,
		state.NewStateHolder(
			state.NewState(
				b.pendingBlock.ledgerView,
				meter.NewMeter(math.MaxUint64, math.MaxUint64),
				state.WithMaxKeySizeAllowed(b.vmCtx.MaxStateKeySize),
				state.WithMaxValueSizeAllowed(b.vmCtx.MaxStateValueSize),
				state.WithMaxInteractionSizeAllowed(b.vmCtx.MaxStateInteractionSize),
			),
		),
		programs.NewEmptyPrograms(),
		flow.NewTransactionBody(),
		0,
		nil)
```

Reply:
PS: Not that I like this way, normally fvm should expose this storage ( from cadence side it is designed fvm as consumer, rightfully )

Reply:
Ohhh right, this is so much better. Didn't even know FVM exposes this. My bad, thank you for this advice!